src/job_cache/eviction_policy.cpp

```diff
@@ -213,9 +222,10 @@ struct LRUEvictionPolicyImpl {
     // Now we remove the threads in the background.
     // TODO: Figure out how many cores we actully have and use a multiple of that
     // NOTE: We don't wait for this to finish
-    std::async(std::launch::async, [dir = cache_dir, jobs_to_remove = std::move(jobs_to_remove)]() {
-      remove_backing_files(dir, jobs_to_remove, 8, 128);
-    });
+    // std::async(std::launch::async, [dir = cache_dir, jobs_to_remove =
+    // std::move(jobs_to_remove)]() {
```

Comment:
Can you explain / fix the commented code here?

Reply:
Oh right. This is the "async's destructor blocks" issue. I should probably put a bit of work into using an std::thread like I did before instead.